aegisthus-hadoop/src/main/java/com/netflix/aegisthus/io/writable/AegisthusKey.java

```diff
@@ -134,6 +135,30 @@ public void readFields(DataInput dis) throws IOException {
         }
     }
 
+    /**
+     * Zero copy readFields.
+     * Note: As defensive copying is not done, caller should not mutate b1 while using instance.
+     * */
+    public void readFields(byte[] b1, int s1, int l1) {
+        int keyLength = WritableComparator.readInt(b1, s1);
+        this.key = ByteBuffer.wrap(b1, s1 + 4, keyLength);
```

Comment:
not urgent but might be cleaner to create a variable and then use that instead of adding 4 everywhere